pkg/server/ca/ca_test.go

```diff
@@ -344,6 +350,54 @@ func (s *CATestSuite) TestSignCAX509SVIDValidatesTrustDomain() {
 	s.Require().EqualError(err, `"spiffe://foo.com" is not a member of trust domain "example.org"`)
 }
 
+func (s *CATestSuite) TestAddCheck() {
+	err := s.healthChecker.AddCheck("server.ca", &caHealth{ca: s.ca, td: trustDomainFoo})
+	s.Require().Error(err)
+	s.Require().EqualError(err, `check "server.ca" has already been added`)
+}
+
+func (s *CATestSuite) TestHealthChecks() {
+	// Successful health check
+	s.Equal(map[string]health.State{
+		"server.ca": {
+			Live:         true,
+			Ready:        true,
+			ReadyDetails: caHealthDetails{},
+			LiveDetails:  caHealthDetails{},
+		},
+	}, s.healthChecker.RunChecks())
+
+	// Failed health check (no X509 CA available)
+	s.ca.SetX509CA(nil)
+	s.Equal(map[string]health.State{
+		"server.ca": {
+			Live:  false,
+			Ready: false,
+			ReadyDetails: caHealthDetails{
+				SignX509SVIDErr: "X509 CA is not available for signing",
+			},
+			LiveDetails: caHealthDetails{
+				SignX509SVIDErr: "X509 CA is not available for signing",
+			},
+		},
+	}, s.healthChecker.RunChecks())
+
+}
+
+func (s *CATestSuite) TestReadyState() {
+	ready, details := s.healthChecker.ReadyState()
+
+	s.True(ready, "CA is not ready")
+	s.Require().NotEmpty(details, "details are not provided for CA health")
+}
+
+func (s *CATestSuite) TestLiveState() {
+	ready, details := s.healthChecker.LiveState()
+
+	s.True(ready, "CA is not live")
+	s.Require().NotEmpty(details, "details are not provided for CA health")
+}
```

Comment:
These tests probably belong in the `health` package? We are already asserting that the ServerCA is returning the proper health.State. The ServerCA unit tests shouldn't be concerned about how the health checker translates the `health.State` into ready and live states.

Reply:
I'll write the unit tests for the health subsystem in subsequent commits.